mm2src/mm2_event_stream/src/lib.rs

```diff
@@ -0,0 +1,56 @@
+use serde::Deserialize;
+
+/// multi-purpose/generic event type that can easily be used over the event streaming
+pub struct Event {
+    _type: String,
+    message: String,
+}
+
+impl Event {
+    pub fn new(event_type: String, message: String) -> Self {
+        Self {
+            _type: event_type,
+            message,
+        }
+    }
+
+    pub fn event_type(&self) -> &str { &self._type }
+
+    pub fn message(&self) -> &str { &self.message }
+}
+
+/// Configuration for event streaming
+#[derive(Deserialize)]
+pub struct EventStreamConfiguration {
+    #[serde(default)]
+    pub access_control_allow_origin: String,
+    #[serde(default)]
+    pub active_events: Vec<EventStatus>,
```

Comment:
How about making this a `HashMap`? configuration wouldn't allow a duplicate event in this case, e.g.
```json
"event_stream_configuration": {
    "access_control_allow_origin": "*",
    "active_events": {
        "NETWORK": {
            "stream_interval_seconds": 0.5
        }
    }
}
```